demos/minimal/actors/sample_generator.py

```diff
@@ -15,54 +15,52 @@ class Generator(Actor):
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        self.data = None
         self.name = "Generator"
-        self.frame_num = 0
+        self.frame_num = 0  # Initialize frame counter
+        # Generate xs (x-coordinates)
+        self.xs = np.linspace(-10, 10, 100)
 
     def __str__(self):
-        return f"Name: {self.name}, Data: {self.data}"
+        return f"Name: {self.name}"
 
     def setup(self):
-        """Generates an array that serves as an initial source of data.
-
-        Initial array is a 100 row, 5 column numpy matrix that contains
-        integers from 1-99, inclusive.
-        """
-
+        """Initial setup for Generator."""
         logger.info("Beginning setup for Generator")
-        self.data = np.asmatrix(np.random.randint(100, size=(100, 5)))
         logger.info("Completed setup for Generator")
 
     def stop(self):
-        """Save current randint vector to a file."""
-
+        """Actions to perform on stopping."""
         logger.info("Generator stopping")
-        np.save("sample_generator_data.npy", self.data)
         return 0
 
     def runStep(self):
-        """Generates additional data after initial setup data is exhausted.
-
-        Data is of a different form as the setup data in that although it is
-        the same size (5x1 vector), it is uniformly distributed in [1, 10]
-        instead of in [1, 100]. Therefore, the average over time should
-        converge to 5.5.
-        """
+        """Sends a dictionary containing frame number and 2D array with x and y coordinates."""
+        time.sleep(0.5)  # Delay for half a second
 
-        if self.frame_num < np.shape(self.data)[0]:
-            data_id = self.client.put(
-                self.data[self.frame_num], str(f"Gen_raw: {self.frame_num}")
-            )
-            # logger.info('Put data in store')
-            try:
-                self.q_out.put([[data_id, str(self.frame_num)]])
-                logger.info("Sent message on")
-                self.frame_num += 1
-            except Exception as e:
-                logger.error(
-                    f"--------------------------------Generator Exception: {e}"
-                )
+        # Generate sine or cosine values based on frame number
+        if self.frame_num % 2 == 0:
+            # Even frame: Generate sine wave
+            ys = np.sin(self.xs)
         else:
-            self.data = np.concatenate(
-                (self.data, np.asmatrix(np.random.randint(10, size=(1, 5)))), axis=0
-            )
+            # Odd frame: Generate cosine wave
+            ys = np.cos(self.xs)
+
+        # Combine x and y into a 2D array
+        values = np.column_stack((self.xs, ys))  # Shape (100, 2)
+
```

Comment:
For below this line: see how the current sample generator `runStep` is organized and you should mimic: https://github.com/project-improv/improv/blob/dev/demos/minimal/actors/sample_generator.py

Again, we should not be putting the data in a dictionary. The preference is always going to be passing things around in a numpy array and then unpacking it using indexing.